hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetCommandLine.java

```diff
@@ -712,4 +814,373 @@ static Throwable peel(Throwable e) {
+                throw new EOFError(-1, quoteStart, "Missing closing quote",
+                        line.charAt(quoteStart) == '\'' ? "quote" : "dquote");
+            }
+
+            if (multiLineCommentStart != -1) {
+                throw new EOFError(-1, cursor, "Missing end of comment", "**");
+            }
+
+            final int lastNonQuoteCommentIndex1 =
+                    lastNonQuoteCommentIndex == line.length() - 1
+                            && lastNonQuoteCommentIndex - 1 >= 0
+                            ? lastNonQuoteCommentIndex - 1 : lastNonQuoteCommentIndex;
+            if (containsNonCommentData
+                    && !isLineFinishedWithSemicolon(
+                    lastNonQuoteCommentIndex1, line)) {
+                throw new EOFError(-1, cursor, "Missing semicolon (;)");
+            }
+        }
+
+        /**
+         * Returns whether a line (already trimmed) ends with a semicolon that
+         * is not commented with one line comment.
+         *
+         * <p>ASSUMPTION: to have correct behavior, this method must be
+         * called after quote and multi-line comments check calls, which implies that
+         * there are no non-finished quotations or multi-line comments.
+         *
+         * @param buffer Input line to check for ending with ';'
+         * @return true if the ends with non-commented ';'
+         */
+        private boolean isLineFinishedWithSemicolon(
+                final int lastNonQuoteCommentIndex, final CharSequence buffer) {
+            final String line = buffer.toString();
+            boolean lineEmptyOrFinishedWithSemicolon = line.isEmpty();
+            boolean requiredSemicolon = false;
+            String[] oneLineComments = {"#", "--"};
+            for (int i = lastNonQuoteCommentIndex; i < line.length(); i++) {
+                if (';' == line.charAt(i)) {
+                    lineEmptyOrFinishedWithSemicolon = true;
+                    continue;
+                } else if (i < line.length() - 1
+                        && line.regionMatches(i, "/*", 0, "/*".length())) {
+                    int nextNonCommentedChar = line.indexOf("*/", i + "/*".length());
+                    // From one side there is an assumption that multi-line comment
+                    // is completed, from the other side nextNonCommentedChar
+                    // could be negative or less than lastNonQuoteCommentIndex
+                    // in case '/*' is a part of quoting string.
+                    if (nextNonCommentedChar > lastNonQuoteCommentIndex) {
+                        i = nextNonCommentedChar + "*/".length();
+                    }
+                } else {
+                    for (String oneLineCommentString : oneLineComments) {
+                        if (i <= buffer.length() - oneLineCommentString.length()
+                                && oneLineCommentString
+                                .regionMatches(0, line, i, oneLineCommentString.length())) {
```

Comment:
I think the `toffset` and `ooffset` are swapped here, it should have been:
```
.regionMatches(i, line, 0, oneLineCommentString.length()))
```
But using `startsWith` will be much more readable IMO:
```suggestion
                        if (line.substring(i).startsWith(oneLineCommentString)) {
```
Same for `regionMatches` above.

There also seems to be duplicate code in `isOneLineComment` below.

Reply:
Number should be right-aligned.
![image](https://user-images.githubusercontent.com/9109877/102804191-b2a37680-43b9-11eb-9bf9-f7454bacca1c.png)

Reply:
I removed the inconsistent regionMatches usages and followed a single style as line.regionMatches(...)